src/java/org/apache/cassandra/db/Keyspace.java

```diff
@@ -373,6 +375,11 @@ public Row getRow(QueryFilter filter)
     {
         ColumnFamilyStore cfStore = getColumnFamilyStore(filter.getColumnFamilyName());
         ColumnFamily columnFamily = cfStore.getColumnFamily(filter);
+        int readDelay = DatabaseDescriptor.getReadDelay();
```

Comment:
Might make sense to leave this from this PR; basically the read delay being set is only network level, and is not storage level. Thus, dynamic snitch can never function properly...

Reply:
This fixes that by making the pause in the read code

Reply:
I'm moving this in https://github.com/palantir/cassandra/pull/232